### ISSUE TITLE
Table 'wrhrankmain' doesn't exist

### ISSUE THREAD

**Issue by CustomStats** [shavitush/bhoptimer#1056]
Every other new table seems to have been created just fine (ex: wrs, wrs_min, stagetimeswr) but this one doesn't exist. Not sure if this is something we're supposed to add manually or if it was supposed to be automated (i don't know db shit), so let me know if this is on us.

```
[shavit-rankings.smx] SQL_GetWRs_Callback failed. Reason: Table 'sourcemod_bhoptimer_dev.wrhrankmain' doesn't exist
```

**Comment by rtldg** [comment]
It's a temporary timer created by the timer.
What version of mysql / mariadb are you using?
And also does this happen often or was it just a one-off error?

**Comment by CustomStats** [comment]
Happens whenever someone joins the server. 
```
L 06/06/2021 - 22:38:02: "Custom Stats<12><STEAM_1:0:89886072><>" entered the game
L 06/06/2021 - 22:38:02: [shavit-rankings.smx] SQL_GetWRs_Callback failed. Reason: Table 'sourcemod_bhoptimer_dev.wrhrankmain' doesn't exist
```

"10.4.17-MariaDB-1:10.4.17+maria~focal"

**Comment by rtldg** [comment]
I think this is happening because the SQL user does not have the `CREATE TEMPORARY TABLES` privilege.
I'm able to replicate it if I revoke the privilege from my user with `REVOKE CREATE TEMPORARY TABLES ON dbname.* FROM foo2@localhost; FLUSH PRIVILEGES;`
Adding the privilege can be done with something like `GRANT CREATE TEMPORARY TABLES ON dbname.* TO foo2@localhost; FLUSH PRIVILEGES;`
also could use `*.*` instead of a single database

**Comment by rtldg** [comment]
[made the `wrhrank*` temp tables be VIEWs instead](https://github.com/shavitush/bhoptimer/commit/d4b61a474f2d77b35ff5ce07c630e406e8fce1cb) so this should be solved for 3.0.5. Let me know if any problems come up again
